DATABASE_MANAGEMENT.md

```diff
@@ -82,6 +82,43 @@ A node in this case is an XML node like an element, attribute, text, etc.
 ### Sources
 http://www.adamretter.org.uk/presentations/restxq_mugl_20120308.pdf
 
+## Database Rights
+In a microservice architecture, a microservice's database should be part of the implementation of that service and cannot be accessed directly by other services. This way, the service is loosely coupled and can be developed/scaled/deployed independently.
+
+There are some patterns to keep persistent data private:
+- private-tables-per-service
+- schema-per-service
+- database-server-per-service
+As seen, 2 options are not available for BaseX because it's not a relational database. It doesn't have tables or schemas.
+A Database-server-per-service pattern helps ensure that the services are lossely coupled.
+
+The CIM - IEC 61850 service for example get's their own database. If another service wants to get SCD files from this service, use the API of that particular service.
+
+### Where do we set the user privelages of Basex?
+Basex has it's own [User Management](https://docs.basex.org/wiki/User_Management).
+
+It's pretty straight forward: Basex has Users that can be created. These users can have so-called permissions that can be applied to the user:
+![BaseX permissions overview](./images/database/basex_permissions.png)
+
+In this overview, we see 'Global' permissions and 'Local' permissions.
+In both permission groups, a higher permission includes all lower permissions. So a user with the 'Create' permission also has the 'Read' permission.
+
+All permissions are stored in a file called users.xml (which can be editted manually) inside the database directory, and is being parsed once BaseX is started.
+
+### How do we connect BaseX with a central identity repository/application?
+BaseX doesn't have compatibility with a central identity repository (like Keycloak) out of the box available, but after discussing it with the BaseX community it's pretty comfortable to achieve this with RESTXQ or xQuery. There are examples available for making use of Keycloak:
+
+[Example with xQuery](https://code-repo.d4science.org/gCubeSystem/d4science-keycloak-themes/src/branch/master/src/utils/xquery)
+Example with RESTXQ is not available yet. The RESTXQ module, in combination with the basex:perm annotation, protects the access to a GUI implementing the OAuth2 Code-grant flow. This is definitely something we can use.
+
+### Is direct database access allowed within the microservices architecture?
+For maintenance for example, it's of course allowed to have direct database access. There is no best practice available for this. For some things, you just need direct database access.
+
+If other microservices need access to the data of an other microservice, the only way (best practice) to do this is by API calls.
+
+Source:
+https://microservices.io/patterns/data/database-per-service.html
```

Comment:
"Each service has database credentials that only grant it access its own (logical) database on a shared MySQL server."

Is this the setup we want to do here with BaseX?

Reply:
No, each service has it's own database.

But parts of the database can be enclosed for certain API calls, which can be done by credentials / certain users.